_transcripts/039-rust-1.54-1.55.md

```diff
@@ -0,0 +1,718 @@
+target families before, if you've ever written something like, conditional
+compilation thing, like `#[cfg(...)]` and then like `unix` or `windows`, you've
+used target families. So target families are collections of target operating
+systems. So for example, the unix target family, which `unix` is a shorthand
+for, includes Linux, but it also includes Android, for example. Similarly,
+`windows` includes all the different versions of Windows. And now `wasm`
+includes all the different types of wasm that there are, they all grouped under
+the target family of `wasm`. It's sort of a collector for targets.
+
+We got another thing in 1.54, which is, we finally got incremental compilation
+back. So you'll remember that in the previous episode, we talked about how
+incremental compilation was disabled, because there were some corner cases that
+weren't handled correctly by the compiler. And those used to just be silently
+ignored. And now when they sort of enabled the way to actually detect them, they
+made them fatal errors, sort of by accident. And then they decided they actually
+should be. Now all of the relevant compiler errors that have been detected by
+that sort of incremental compilation sanity checking, all of those have now been
+fixed and so they've deemed that incremental compilation can now be re-enabled
+by default. There are still a couple I think minor ICEs that are very rare that
+are still being tracked, but it was— they sort of made the decision that at this
+point it's correct to turn it back on, and basically no one will run into the
+corner cases. Happy to get this back.
+
+__Ben__: Because this release came out about six or so weeks ago— longer, by the
+time you listen to this. We did go through the issue and— to see if anyone has
+registered any kind of new complaints, to see if maybe, like, you know, there
+was, it was premature, but it seems like there was pretty much nobody who has
+been noticing. At least not reporting any new problems with this, turning this
+back on. So hopefully, it all should be pretty good. And I think, I believe I
+saw a comment from one of the developers working on this, which is that even all
+the minor bugs should be resolved as of a recent nightly. So...
+
+__Jon__: It's exciting. It's almost like things are getting better.
+
+__Ben__: Well, a great way to make things better is to make them worse and then
+they get better.
+
+__Jon__: We're actually almost at the end of this release, so I guess you were
+right that it's fairly short. Although we do have some stabilized APIs and some
+some more detailed changelog items to go through. For stabilized APIs we already
+talked about the wasm32 intrinsics, which have gotten their own module under
+arch called `wasm32`. There's some binary search things added to `VeqDeque`— or
+*vec-deque*, I don't know how it's pronounced— that we're not really going to
+talk about; they're not super interesting.
```

Comment:
I'm always a bit unsure how to transcribe debates about how something is pronounced...

Reply:
Hehe, yeah, I don't have a good answer for you. I suppose we could use the phonetic alphabet (IPA) for this to describe the sounds, but that _feels_ like too much work.

Reply:
Alternatively write "pronounced like Vec Deck" and "pronounced like Vec D Q"?

Reply:
I'm not in love with any of the alternatives, so I just left it as-is?